code/game/machinery/doors/_door.dm

```diff
@@ -129,11 +122,24 @@
 		return 0
 	return 1
 
+/obj/machinery/door/proc/set_bounds()
+	if (dir == NORTH || dir == SOUTH)
```

Comment:
You should prune diagonal dirs put somewhere so this only sets on cardinal values.